src/main.cpp

```diff
@@ -783,6 +793,31 @@ int main ( int argc, char** argv )
                 strServerBindIP = "";
             }
         }
+
+#ifndef NO_JSON_RPC
+        //
+        // strJsonRpcBind address defaults to loopback and should not be empty, but
+        // in the odd chance that an empty IP is passed, we'll check for it here.
+
+        if ( strJsonRpcBindIP.trimmed().isEmpty() )
+        {
+            qCritical() << qUtf8Printable ( QString ( "JSON-RPC is enabled but no bind address was provided, exiting." ) );
```

Comment:
I think the wording of the "objection" is slightly misleading here.  `--serverbindip` was passed with a parameter, otherwise `GetStringArgument` would have failed earlier.  So
```suggestion
            qCritical() << qUtf8Printable ( QString ( "JSON-RPC is enabled but bind address provided is invalid, exiting." ) );
```
It might even be useful to do `if ( !InetAddr.setAddress ( strJsonRpcBindIP ) )`, like for Server Bind IP.

I'd also question whether "exiting" is in keeping with the other checks we do... it's a bit hit and miss, I guess.  I may even have said do it like this :).

Reply:
> I think the wording of the "objection" is slightly misleading here.

I'll squeeze that improvement into #2958 and rename it to be more generic.

> I'd also question whether "exiting" is in keeping with the other checks we do... it's a bit hit and miss, I guess. I may even have said do it like this :).

I would have certainly supported that (maybe I did ;)). I think we should fail early for new features, even if we didn't do so previously and don't want to make those existing checks more strict for compatibility reasons for now.